internal/controller/processor.go

```diff
@@ -70,6 +70,9 @@ func (processor *AppConfigurationProviderProcessor) processFullReconciliation()
 		return err
 	}
 	processor.Settings = updatedSettings
+	for _, reference := range updatedSettings.SecretReferences {
```

Comment:
Current all the version check are in the processor, result in we need to set the `updateNeeded` everywhere for the secret references, could we put the version check in the controller right before trying to update the Secret? I think that would be more clear.